res/controllers/Pioneer-DDJ-SR2-scripts.js

```diff
@@ -0,0 +1,1411 @@
+// If set to TRACK - sets the color to what the track column says.
+// If set to POSITION - Changes color as playposition changes.
+DDJSR2.wheelColor =[];
+DDJSR2.wheelColor[1] = "TRACK"; // Deck 1
+DDJSR2.wheelColor[2] = "ORANGE"; // Deck 2
+DDJSR2.wheelColor[3] = "CYAN"; // Deck 3
+DDJSR2.wheelColor[4] = "BRIGHTBLUE"; // Deck 4
+
+DDJSR2.wheelLedCircleColor = {
+    OFF: 20,
+    TRACK: 127,
+    POSITION: 128,
+    PURPLE: 1,
+    MAGENTA: 2,
+    RED: 3,
+    BROWN: 4,
+    ORANGE: 5,
+    YELLOW: 6,
+    LIMEGREEN: 7,
+    GREEN: 8,
+    BRIGHTGREEN: 9,
+    CELESTE: 10,
+    CYAN: 11,
+    TURQUOISE: 12,
+    BRIGHTBLUE: 13,
+    AZURE: 14,
+    DARKBLUE: 15,
+    VIOLET: 16,
+    ULTRAVIOLET: 17,
+    WHITE: 18,
+    DIMWHITE: 19
+};
+
+DDJSR2.wheelColorMap = new ColorMapper({
+    0xCC0000: DDJSR2.wheelLedCircleColor.RED,
+    0x008000: DDJSR2.wheelLedCircleColor.GREEN,
+    0x0000CC: DDJSR2.wheelLedCircleColor.BLUE,
+    0xFFFF00: DDJSR2.wheelLedCircleColor.YELLOW,
+    0x0088CC: DDJSR2.wheelLedCircleColor.CELESTE,
+    0xFF8800: DDJSR2.wheelLedCircleColor.ORANGE,
+    0x895129: DDJSR2.wheelLedCircleColor.BROWN,
+    0xFF00FF: DDJSR2.wheelLedCircleColor.MAGENTA,
+    0xEE82EE: DDJSR2.wheelLedCircleColor.VIOLET,
+    0x8800CC: DDJSR2.wheelLedCircleColor.PURPLE,
+    0xFFFFFF: DDJSR2.wheelLedCircleColor.WHITE,
+});
+
+
+////////////
+// Code. //
+//////////
+
+DDJSR2.seratoHeartbeat=[0xF0, 0x00, 0x20, 0x7F, 0x50, 0x01, 0xF7]; // Serato Heartbeat
+DDJSR2.seratoControlsPoll=[0xF0, 0x00, 0x20, 0x7F, 0x03, 0x01, 0xF7]; // Serato first poll
+
+DDJSR2.init = function() {
+    // Browser section is part of Mixer container for now.
+
+    DDJSR2.browser = new DDJSR2.BrowserContainer();
+
+    DDJSR2.deck = [];
+    for (let i = 0; i < 4; i++) {
+        DDJSR2.deck[i] = new DDJSR2.Deck(i);
+        DDJSR2.deck[i].setCurrentDeck(`[Channel${i+1}]`);
+    }
+
+    DDJSR2.mixer = new DDJSR2.MixerContainer();
+
+    DDJSR2.createEffectUnits();
+
+    // Performance pads are part of the Deck specs, as are Illumination Controls
+
+    DDJSR2.seratoHeartbeatTimer = engine.beginTimer(200, DDJSR2.doSeratoHeartbeatTimer, false);
+
+    // After 500 ms have elapsed, after the serato heartbeat has occurred, then poll controls. Calling this before time results in an unsuccessful poll.
+
+    engine.beginTimer(300, function() {
+        midi.sendSysexMsg(DDJSR2.seratoControlsPoll, DDJSR2.seratoControlsPoll.length);
+    }, true);
+
+};
+
+DDJSR2.doSeratoHeartbeatTimer = function() {
+    midi.sendSysexMsg(DDJSR2.seratoHeartbeat, DDJSR2.seratoHeartbeat.length);
+};
+
+
+// First, a function library.
+////////////////
+
+
+DDJSR2.getRotaryDelta = function(value) {
+    let delta = 0x40 - Math.abs(0x40 - value);
+    const isCounterClockwise = value > 0x40;
+    // BrowserKnob returns 1-30 going down, and 127-98 going up.
+
+    if (isCounterClockwise) {
+        delta *= -1;
+    }
+    return delta;
+};
+
```

Comment:
right, that can be done way simpler...
```suggestion
DDJSR2.getRotaryDelta = function(value) {
    return value > 0x40 : value - 0x7F : value;
};

```